src/components/smartSearch/filters/TimeFrame.tsx

```diff
@@ -0,0 +1,120 @@
+                return;
+            }
+            else if (timeFrame === TIME_FRAME.BETWEEN && after && before) {
+                setAfter(new Date(after));
+                setBefore(new Date(before));
+            }
+            else if (timeFrame === TIME_FRAME.AFTER_DATE && after) {
+                setAfter(new Date(after));
+            }
+            else if (timeFrame === TIME_FRAME.BEFORE_DATE && before) {
+                setBefore(new Date(before));
+            }
+        }
+    }, []);// eslint-disable-line react-hooks/exhaustive-deps
+    useEffect(() => {
+        if (selected ===  TIME_FRAME.EVER) {
+            onChange({});
+        }
+        if (selected === TIME_FRAME.BEFORE_TODAY) {
+            onChange({ before: 'now' });
+        }
+        if (selected === TIME_FRAME.FUTURE) {
+            onChange({ after: 'now' });
+        }
+        if (selected === TIME_FRAME.LAST_FEW_DAYS) {
+            onChange({ after: `-${numDays}d` });
+        }
+        if (selected === TIME_FRAME.BEFORE_DATE) {
+            onChange({ before: before.toISOString().slice(0, 10) });
+        }
+        if (selected === TIME_FRAME.AFTER_DATE) {
+            onChange({ after: after.toISOString().slice(0, 10) });
+        }
+        if (selected === TIME_FRAME.BETWEEN) {
+            onChange({
+                after: after.toISOString().slice(0, 10),
+                before: before.toISOString().slice(0, 10),
+            });
+        }
+    }, [before, after, selected, numDays]); // eslint-disable-line react-hooks/exhaustive-deps
+
+    return (
+        <MuiPickersUtilsProvider utils={ DateFnsUtils }>
+            <Typography display="inline" variant="h4">
+                <Msg id={ `misc.smartSearch.timeFrame.edit.${selected}` } values={{
+                    afterDateSelect: (
+                        <StyledDatePicker
+                            onChange={ (date) => setAfter(date as Date) }
+                            value={ after }
+                        />
+                    ),
+                    beforeDateSelect: (
+                        <StyledDatePicker
+                            onChange={ (date) => setBefore(date as Date) }
+                            value={ before }
+                        />
+                    ),
+                    daysInput: (
+                        <StyledNumberInput
+                            onChange={ (e) => setNumDays(+e.target.value) }
+                            value={ numDays }
+                        />
+                    ),
+                    timeFrameSelect: (
+                        <StyledSelect
+                            onChange={ e => setSelected(e.target.value as TIME_FRAME) }
+                            SelectProps={{ renderValue: function getLabel(value) {
+                                return (
+                                    <Msg
+                                        id={ `misc.smartSearch.timeFrame.timeFrameSelectLabel.${value}` }
```

Comment:
This message requires the `days: numDays` parameter to correctly render the translation in some languages, e.g. Swedish.